packages/node/src/utils/cosmos.ts

```diff
@@ -184,21 +194,11 @@ export function wrapEvent(
     try {
       logs = parseRawLog(tx.tx.rawLog) as Log[];
     } catch (e) {
+      logger.warn(e, 'Faied to parse raw log');
       continue;
     }
     for (const log of logs) {
-      const decodedMsg = api.decodeMsg(
-        tx.decodedTx.body.messages[log.msg_index],
-      );
-      const msg: CosmosMessage = {
-        idx: log.msg_index,
-        tx: tx,
-        block: block,
-        msg: {
-          typeUrl: tx.decodedTx.body.messages[log.msg_index].typeUrl,
-          ...decodedMsg,
-        },
-      };
+      const msg = wrapCosmosMsg(block, tx, log.msg_index, api);
```

Comment:
We should be smarter about this. We already do the same conversion in `wrapMsg` and we can end up doing the same message in this function too.